app/controllers/user.js

```diff
@@ -44,6 +44,46 @@ exports.create = (req, res, next) => {
   
 };
 
+exports.signin = (req, res, next) => {
+
+  let input = emptyToNull(req.body);
+
+  if(!input.email || !input.password){
+    return res.status(400).send('Both email and password are required to continue');
+  }
+  if (req.session && req.session.user && req.session.user.email === input.email){
+    return res.status(200).send('You are already logged in!');
+    
+  }
+
+  User.findOne({
+    where: {
+      email: input.email
+    }
+  }).then(result => {
+    
+    if(!result || !result.email){
```

Comment:
If you have already checked the presence of the email param above, you can only ask for the presence of result.